promql/parser/ast.go

```diff
@@ -125,10 +125,17 @@ type MatrixSelector struct {
 
 // SubqueryExpr represents a subquery.
 type SubqueryExpr struct {
-	Expr   Expr
-	Range  time.Duration
-	Offset time.Duration
-	Step   time.Duration
+	Expr  Expr
+	Range time.Duration
+	// OriginalOffset is the actual offset that was set in the query.
+	// This never changes.
+	OriginalOffset time.Duration
+	// Offset is the offset used during the query execution
+	// which is calculated using the original offset, at modifier time,
+	// eval time, and subquery offsets in the AST tree.
+	Offset    time.Duration
```

Comment:
Ooops, changing the meaning of an existing field without renaming / removing it caught me off guard and introduced a bug in PromLens where it now thinks all offsets are 0 (because `Offset` is now always 0 after parsing). I know we don't have API stability guarantees, but in the future I would recommend in this type of situation to do it the other way around: keep the same semantics for the existing field and instead add another one for the new meaning (e.g. `AdjustedOffset` or something like that).